src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -201,14 +202,42 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
             return validName;
         }
 
-        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(Type type)
+        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(
+            Type type, 
+            ErrorSink errorSink)
         {
-            var properties = type.GetRuntimeProperties().Where(IsValidProperty);
+            var properties = type.GetRuntimeProperties().Where(IsAccessibleProperty);
             var attributeDescriptors = properties.Select(ToAttributeDescriptor);
 
+            attributeDescriptors = attributeDescriptors.Where(
+                descriptor => ValidTagHelperAttributeDescriptor(descriptor, type, errorSink));
+
             return attributeDescriptors;
         }
 
+        private static bool ValidTagHelperAttributeDescriptor(
```

Comment:
why isn't this part of the existing `IsValid` method?  nesting the `where` clauses instead of joining the two conditions just to avoid calculating the name twice?  the extra complication (and state machine) seems like the wrong trade-off.